crates/runtime/src/datafusion/query/cache.rs

```diff
@@ -102,9 +102,9 @@ impl Query {
         };
 
         // Use the logical plan with parameter values for caching and lookup
-        let plan = match parameters {
+        let plan = match &parameters {
             Some(param_values) => plan
-                .with_param_values(param_values)
+                .with_param_values(param_values.clone())
```

Comment:
Is this a true clone or just a reference counter increase? If the previous lets refactor this so that we're passing pointers to the value instead of cloning it